From david-b@pacbell.net  Sun Jan 25 12:54:56 2009
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Sat, 24 Jan 2009 17:57:06 -0800
Subject: USB: prevent musb init oops (otg related)
To: linux-usb@vger.kernel.org
Cc: Greg KH <greg@kroah.com>, Felipe Balbi <felipebalbi@users.sourceforge.net>, Ajay Kumar Gupta <ajay.gupta@ti.com>
Message-ID: <200901241757.06185.david-b@pacbell.net>
Content-Disposition: inline


From: Ajay Kumar Gupta <ajay.gupta@ti.com>

Subject: usb: musb: fix null pointer check in musb_platform_init()

Fixes kernel panic when musb is selected for OMAP35x EVM but twl4030 is not
selected as in this case otg_get_transceiver() returns null.

Signed-off-by: Ajay Kumar Gupta <ajay.gupta@ti.com>
Acked-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/omap2430.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/musb/omap2430.c
+++ b/drivers/usb/musb/omap2430.c
@@ -231,6 +231,10 @@ int __init musb_platform_init(struct mus
 #if defined(CONFIG_ARCH_OMAP2430)
 	omap_cfg_reg(AE5_2430_USB0HS_STP);
 #endif
+	if (!x) {
+		pr_err("omap: musb: null transceiver found\n");
+		return -ENODEV;
+	}
 
 	musb_platform_resume(musb);
 
